src/gtfs/read.rs

```diff
@@ -199,12 +321,176 @@ pub fn read_stops<P: AsRef<path::Path>>(
     (stopareas, stoppoints)
 }
 
+#[derive(Eq, PartialEq)]
+enum RouteReadType {
+    RouteAsNtmRoute,
+    RouteAsNtmLine,
+}
+
+fn define_route_file_read_mode(gtfs_routes: &Vec<Route>) -> RouteReadType {
+    let mut iter = gtfs_routes.iter();
+    while let Some(r1) = iter.next() {
+        for r2 in iter.clone() {
+            if r1.is_same_line(r2) {
+                return RouteReadType::RouteAsNtmRoute;
+            }
+        }
+    }
+    return RouteReadType::RouteAsNtmLine;
+}
+
+fn get_commercial_mode_label(route_type: &RouteType) -> String {
+    let result = match route_type {
+        &RouteType::Tramway_LightRail => "Tram, Streetcar, Light rail",
+        &RouteType::Metro => "Subway, Metro",
+        &RouteType::Rail => "Rail",
+        &RouteType::Bus => "Bus",
+        &RouteType::Ferry => "Ferry",
+        &RouteType::CableCar => "Cable car",
+        &RouteType::Gondola_SuspendedCableCar => "Gondola, Suspended cable car",
+        &RouteType::Funicular => "Funicular",
+        _ => "Other Mode",
+    };
+    result.to_string()
+}
+
+fn get_commercial_mode(route_type: &RouteType) -> objects::CommercialMode {
+    objects::CommercialMode {
+        id: route_type.to_gtfs_value(),
+        name: get_commercial_mode_label(&route_type),
+    }
+}
+
+fn get_physical_mode(route_type: &RouteType) -> objects::PhysicalMode {
+    match route_type {
+        &RouteType::Tramway_LightRail => objects::PhysicalMode {
+            id: "RailShuttle".to_string(),
+            name: "Navette ferrée (VAL)".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Metro => objects::PhysicalMode {
+            id: "Metro".to_string(),
+            name: "Métro".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Rail => objects::PhysicalMode {
+            id: "Train".to_string(),
+            name: "Train".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Bus => objects::PhysicalMode {
+            id: "Bus".to_string(),
+            name: "Bus".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Ferry => objects::PhysicalMode {
+            id: "Ferry".to_string(),
+            name: "Ferry".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::CableCar => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Gondola_SuspendedCableCar => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Funicular => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
```

Comment:
I have trouble with this one. The borrow checker raise an error on `match *route_type` and if I try to remove `RouteType::` using `use RouteType::*` I have an other weird error `pattern doesn't bind 'Funicular'` like the compiler understand the match arms like variables ... 
I ended combining match arms the long way...
Can you give me a hand on this one ?

Reply:
I found the solution : 
```rust
use self::RouteType::*;
```
I thought using it inside the two functions would be best to avoid any conflict with the `Other(u16)` of the enum.